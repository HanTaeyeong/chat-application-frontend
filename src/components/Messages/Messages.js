import React from 'react';

import ScrollToBottom from 'react-scroll-to-bottom';
import './Messages.scss'

import Message from './Message/Message';

function Messages({ name, messages }) {
    
    return (
        <ScrollToBottom className={'messages'}>
            {messages?.map((message, i) => <div key={message.user + i + message.text}>
                <Message name={name || ""} message={message} />
            </div>)}
        </ScrollToBottom>
    )
}

export default Messages;
