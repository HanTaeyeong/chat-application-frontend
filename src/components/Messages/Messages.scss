.messages {
  display: flex;
    padding: 5% 0;
    overflow: auto;
    flex: auto;
  }