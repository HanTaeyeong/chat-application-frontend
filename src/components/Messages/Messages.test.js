import React from 'react'
import Messages from './Messages'

import {create} from 'react-test-renderer'

describe('Messages Component Test', () => {
    test('render test', () => {   
        let app=create(<Messages />) 
        expect(app.toJSON()).toMatchSnapshot()
    })
})