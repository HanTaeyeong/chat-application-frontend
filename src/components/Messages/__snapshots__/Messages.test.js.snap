// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Messages Component Test render test 1`] = `
<div
  className="react-scroll-to-bottom--css-klayn-79elbk messages"
>
  <div
    className="react-scroll-to-bottom--css-klayn-1n7m0yu"
  />
</div>
`;
